sql/hive/compatibility/src/test/scala/org/apache/spark/sql/hive/execution/HiveCompatibilitySuite.scala

```diff
@@ -59,6 +59,8 @@ class HiveCompatibilitySuite extends HiveQueryFileTest with BeforeAndAfter {
     TestHive.setConf(SQLConf.IN_MEMORY_PARTITION_PRUNING, true)
     // Ensures that cross joins are enabled so that we can test them
     TestHive.setConf(SQLConf.CROSS_JOINS_ENABLED, true)
+    // Force type case during table insertion
+    TestHive.setConf(SQLConf.LEGACY_INSERT_TABLE_TYPE_COERCION, true)
```

Comment:
Why is this required?

Reply:
Not sure about that. Actually, I don't know how to run the test suite

Reply:
This test is to make sure Spark has the same behavior with Hive, however Hive always add unsafe cast, so we need to turn on this config to make tests pass.

Reply:
Hive automatically inserts unsafe casts?